server/neptune/gateway/event/check_suite_handler.go

```diff
@@ -0,0 +1,88 @@
+package event
+
+import (
+	"context"
+	"github.com/pkg/errors"
+	"github.com/runatlantis/atlantis/server/core/config/valid"
+	"github.com/runatlantis/atlantis/server/events/models"
+	"github.com/runatlantis/atlantis/server/logging"
+	contextInternal "github.com/runatlantis/atlantis/server/neptune/context"
+	"github.com/runatlantis/atlantis/server/neptune/workflows"
+	"github.com/runatlantis/atlantis/server/vcs"
+	"github.com/runatlantis/atlantis/server/vcs/provider/github"
+)
+
+type CheckSuite struct {
+	Action            CheckRunAction
+	HeadSha           string
+	Repo              models.Repo
+	Sender            models.User
+	InstallationToken int64
+	Branch            string
+}
+
+type CheckSuiteHandler struct {
+	Logger            logging.Logger
+	RootConfigBuilder rootConfigBuilder
+	Scheduler         scheduler
+	DeploySignaler    deploySignaler
+}
+
+func (h *CheckSuiteHandler) Handle(ctx context.Context, event CheckSuite) error {
+	if event.Action.GetType() != ReRequestedActionType {
+		h.Logger.DebugContext(ctx, "ignoring checks event that isn't a rerequested action")
+		return nil
+	}
+	// Block force applies
+	if event.Branch != event.Repo.DefaultBranch {
+		h.Logger.DebugContext(ctx, "dropping event branch unexpected ref")
+		return nil
+	}
+	return h.Scheduler.Schedule(ctx, func(ctx context.Context) error {
+		return h.handle(ctx, event)
+	})
+}
+
+func (h *CheckSuiteHandler) handle(ctx context.Context, event CheckSuite) error {
+	builderOptions := BuilderOptions{
```

Comment:
Seeing the same code here: 
https://github.com/lyft/atlantis/blob/61557a2da5a093d420618b3eba1ee49a093964c1/server/neptune/gateway/event/push_event_handler.go#L86

Can we dedup?